lib/LTP/WhiteList.pm

```diff
@@ -124,12 +131,22 @@ sub is_test_disabled {
 
 sub find_whitelist_testsuite {
     my ($suite) = @_;
+    my $issues;
 
-    my $path = get_var('LTP_KNOWN_ISSUES');
+    my $path = get_var('LTP_KNOWN_ISSUES_LOCAL');
     return undef unless defined($path) and -e $path;
 
     my $content = path($path)->slurp;
-    my $issues = Mojo::JSON::decode_json($content);
+
+	if ($content =~ m/^\x{FEFF}*[^:ascii:]*\{/) {
+        $issues = Mojo::JSON::decode_json($content);
+	} elsif ($content =~ m/^\x{FEFF}*[^:ascii:]*---/) {
+        my $yp = YAML::PP->new(schema => [qw/ + Merge /]);
+        $issues = $yp->load_string(decode_utf8 $content);
```

Comment:
If this is to check whether the input is JSON or YAML and cannot be determined from the file extension, then reading everything with YAML::PP might be the better way, as JSON is a subset of YAML.
There are however two things that YAML::PP doesn't support yet:
* Byte Order Marks
* JSON Surrogate Pairs

Reply:
In the end I chose this solution, i.e. parse everything with `YAML::PP`, but before remove BOM.
I tested it various crafted files (BOM, spaces):
https://raw.githubusercontent.com/pevik/kernel-qe/bom/bom-space.json
https://raw.githubusercontent.com/pevik/kernel-qe/bom/bom-space-no-header.yaml